### PR TITLE
Blob detailes validation and suggest available fixes

### PR DESCRIPTION
**Pull request by erfemega**
- Support added for blobDeatiledValidation option
- Support added for suggestAvailableFixes option
- Fix some unit tests
- Now we are handling when there are some elements with the same name in different levels (different parents for each one) with all type of error we support (invalid type, missing in the schema, missing in request, not allowed value).
- Now we are handling when there are some elements with the same name in the same level (all have the same parent), In this case, it is supported with some error types (invalid type, missing in the schema). some error types are not supported yet in this scenario (Missing in request), I did not test not allowed content error yet.
- Validation with schema libraries adapters are now connected with suggesting fix stuff.
- XmllintFacade was modified to use the same XML node (only the functions we are using) format than libxmljs libraries to make easier the connection with this library.
- In xmlLintFacade was added 'xpath' and 'xmldom' libraries to give support to find in a document by xpath.
- In the mismatch suggestion, the suggested value and actual value were modified to contain only the element value when they are an element with text. When they are an element with children the key will corresponds with the element's parent and de suggested and actual values will contain the children elements stringified.

There are two linting errors about a line of ~2500 characters in a test. 
I didn't know how to handle it, I hope @Carlos-Veloz you could help me with those.

I attach the error cases it was built on.
[reasonCodeExamples.md](https://github.com/postmanlabs/wsdl-to-postman/files/6636151/reasonCodeExamples.md)

It could be useful too
https://devhints.io/xpath#prefixes

### CODE REVIEW

**Review by Carlos-Veloz — APPROVED** [review]

File: test/unit/TransactionValidator.options.test.js [hunk 1]
```diff
@@ -368,47 +435,182 @@ describe('validateBody method with options', function () {
...
+      result = transactionValidator.validateTransaction(
+        numberToWordsCollectionItemsBodyMoreFields,
+        numberToWordsWSDLObject, new XMLParser(),
+        {
+          showMissingSchemaErrors: true,
+          suggestAvailableFixes: true,
+          detailedBlobValidation: true
+        }
+      ),
+      mismatchReason = 'Element \'WORNGFIELD\': This element is not expected.\n',
```
Comment (LuisTejedaS) [inline test/unit/TransactionValidator.options.test.js:526]: Maybe should change it to wrongfield

File: lib/TransactionValidator.js [hunk 2]
```diff
@@ -237,6 +242,120 @@ class TransactionValidator {
...
+  getMissingChildKeyFromReason(mismatchReason) {
+    const element = new RegExp(/Expected is \( ([A-Za-z]+) \)/),
+      quotedTextInElement = mismatchReason.match(element)[1];
+    return quotedTextInElement;
+  }
+
+  getElementLineFromBody(body, elementKey) {
+    const tagPattern = new RegExp(`<${elementKey}>[\\s\\S]*?<\/${elementKey}>|<${elementKey}[\\s\\S]*?\/>`),
+      matched = body.match(tagPattern);
+    return matched ? matched[0] : '';
```
Comment (LuisTejedaS) [inline lib/TransactionValidator.js:313]: Are we handling when there is more than one property with the same name? are we going to check that with PM people?

**Review by LuisTejedaS — CHANGES_REQUESTED** [review]
I think you need to add documentation for the methods

File: lib/TransactionValidator.js (hunk 2 above)
Comment (erfemega) [reply lib/TransactionValidator.js]: - Now we are handling when there are some elements with the same name in different levels (different parents for each one) with all type of error we support (invalid type, missing in the schema, missing in request, not allowed value).
- Now we are handling when there are some elements with the same name in the same level (all have the same parent), In this case, it is supported with some error types (invalid type, missing in the schema). some error types are not supported yet in this scenario (Missing in request and I did not test not allowed content yet).

**Review by Carlos-Veloz — APPROVED** [review]

**Review by LuisTejedaS — APPROVED** [review]
LGTM

**Pull request merged**